### ISSUE TITLE
Does not recognize ~/.jscsrc

### ISSUE THREAD

**Issue by scholtzm** [AtomLinter/linter-jscs#75]
I just updated to v2.0.2 from v1.13.0 and it's no longer picking up my ~/.jscsrc. This definitely worked couple hours ago. :grinning:

Is there anything I can do to fix this?

**Comment by rachel-carvalho** [comment]
Same here!

**Comment by iam4x** [comment]
Could you paste your `~/.jscsrc` please?

**Comment by 1000ch** [comment]
Some possibility, `package.json` is detected earlier than `~/.jscsrc`.

**Comment by scholtzm** [comment]
@iam4x I haven't touched my `~/.jscsrc` in weeks but here it is:

```json
{
  "fileExtensions": [".js"],

  "requireParenthesesAroundIIFE": true,
  "validateLineBreaks": "LF",
  "validateIndentation": 4,

  "disallowEmptyBlocks": true
}
```

@1000ch I don't think that's the case since I don't use package.json for the settings.

**btw.** If I check "Only Config - Disable linter if there is no config file found for the linter." in the settings, then it's *not* linting at all. Otherwise it uses the preset.

**Comment by 1000ch** [comment]
@scholtzm 

>Some possibility, package.json is detected earlier than ~/.jscsrc.

In current version of `linter-jscs`, if `package.json` is successfully loaded (whether including `jscsConfig` or not), other config files such like `.jscsrc` and `.jscs.json` are not loaded.

**Comment by scholtzm** [comment]
Yep, sorry I misunderstood your initial comment. It seems like the current code is indeed detecting `package.json` first and using that option even if it does not contain the necessary settings.

**Comment by Eschon** [comment]
:+1:
When I have a `package.json` in my project `.jscsrc` is ignored.
If I remove the `package.json` everything works as expected.

**Comment by piotr-cz** [comment]
So required behaviour is to check package.json and load config from `jscsConfig` key, and when it doesn't exist resolve to `'.jscsrc', '.jscs.json'`

**Comment by 1000ch** [comment]
#76 will resolve this problem!

**Comment by Eschon** [comment]
Ok somehow removing my `package.json` also stopped wokring.
For now I had to disable `linter-jscs` since it seemed to ignore my `.jscsrc` and even the preset option in the settings.

**Comment by arestov** [comment]
it because of `findFile` behavior

https://github.com/atom-community/linter/blob/9950f66a490b3574b19ea6e32f0926e14d4490ce/lib/util.coffee#L21

```
 if nameType is '[object Array]'
      for n in name
        target = path.join(dir, n)

        if fs.existsSync(target)
          if parent
            return dir
          return target
``` 

if `findFile` finds any file in local dir `findFile` returns it.

but it should find each file in all folders from down to up.


So currently linter-jscs finds package.json which does not contain config  (my .jscsrc file is in upper folder), so  linter-jscs uses default config.

____
so we can replace ```const config = findFile(filePath, configFiles);```
with (not familiar with es6)

```
const config = configFiles.reduce(function(result, configPath) {
  return result || findFile(filePath, configPath);
});
```

**Comment by arestov** [comment]
Oh, looks like neither https://github.com/AtomLinter/linter-jscs/pull/76/ nor https://github.com/AtomLinter/linter-jscs/pull/77 fixes this issue in corrent way. 
findFile can't be use here since we can package.json without jscs options in parent folders and not .jscsrc, but can get on top .jscsrc on topper levels or can get package.json with jscs options

**Comment by arestov** [comment]
proper way https://github.com/Joezo/atom-jshint/blob/master/lib/atom-jshint.js#L186

**Comment by iam4x** [comment]
I think we should adress this on atom-linter package, to add options on `findFile` function which will look into home directory.

**Comment by scholtzm** [comment]
Temporary hotfix would be awesome though.

**Comment by iam4x** [comment]
Should be fixed with 48f1ed539e5b212fd95b988f4fd15619e853c17e and with new version published, can you confirm?

**Comment by scholtzm** [comment]
`2.0.3` works correctly for me once again. Thanks :+1:

**Comment by rachel-carvalho** [comment]
`v2.0.3` is working for me too. I had the same problem: `package.json` + `.jscsrc`.

Thank you!

**Comment by iam4x** [comment]
Good :)

**Comment by joshdick** [comment]
Running `2.0.4` here. I have a `package.json` and `.jscsrc` in the root of my project, and the `.jscsrc` doesn't appear to be recognized.

There still appear to be two open pull requests for this issue (#77 and #87).

Is this actually fixed?
